cpp/src/glm/qn/glm_softmax.cuh

```diff
@@ -59,7 +59,11 @@ __global__ void logSoftmaxKernel(T *out, T *dZ, const T *in, const T *labels,
   bool delta = false;
   // TODO is there a better way to read this?
   if (getDerivative && threadIdx.x == 0) {
-    shm.sh_val[threadIdx.y] = labels[y];
+    if (y < N) {
+      shm.sh_val[threadIdx.y] = labels[y];
+    } else {
+      shm.sh_val[threadIdx.y] = std::numeric_limits<T>::lowest();
```

Comment:
I'm sure I'm missing something obvious, but for my own education, can you tell me why we specifically use `lowest` here and not any other value?

Reply:
Of course, because there's a max reduction here https://github.com/rapidsai/cuml/blob/1db5ad2270284b489fe89716fe38cd047843d6dd/cpp/src/glm/qn/glm_softmax.cuh#L89. So these unused values will not interfere with the reduction

Reply:
Aha!!! Of course. Thank you.